openshift/e2e-common.sh

```diff
@@ -111,31 +111,57 @@ function update_csv(){
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.containers[0].volumeMounts.(name==eventing-manifest).mountPath" "/tmp/knative/knative-eventing/0.17.2"
+
+  # volume (serving)
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes[+].name" "serving-manifest"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes.(name==serving-manifest).configMap.name" "ko-data-serving"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes.(name==serving-manifest).configMap.items[+].key" "knative-serving-ci.yaml"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes.(name==serving-manifest).configMap.items.(key==knative-serving-ci.yaml).path" "knative-serving-ci.yaml"
+  # volume (eventing)
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes[+].name" "eventing-manifest"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes.(name==eventing-manifest).configMap.name" "ko-data-eventing"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes.(name==eventing-manifest).configMap.items[+].key" "knative-eventing-ci.yaml"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes.(name==eventing-manifest).configMap.items.(key==knative-eventing-ci.yaml).path" "knative-eventing-ci.yaml"
+
+  # volumeMounts(kourier)
+  yq  write -i ${CSV} 'spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.containers[0].volumeMounts[+].name' kourier-manifest
+  yq  write -i ${CSV} 'spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.containers[0].volumeMounts.(name==kourier-manifest).mountPath' /tmp/kourier
+
+  # volume (kourier)
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.volumes[+].name" "kourier-manifest"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.volumes.(name==kourier-manifest).configMap.name" "kourier-cm"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.volumes.(name==kourier-manifest).configMap.items[+].key" "kourier.yaml"
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.volumes.(name==kourier-manifest).configMap.items.(key==kourier.yaml).path" "kourier.yaml"
+
+  # add KOURIER_MANIFEST env value
+  yq  write -i ${CSV}  "spec.install.spec.deployments.(name==knative-openshift).spec.template.spec.containers.(name==knative-openshift).env.(name==KOURIER_MANIFEST_PATH).value" "/tmp/kourier/kourier.yaml"
```

Comment:
WDYT about moving this all into a `yq` script file? We can either check it in as a separate file or apply it through a string directly. The file could look like this

```
- command: update
  path: spec.install.spec.deployments.(name==knative-operator).spec.template.spec.containers.(name==knative-operator).env[+]
  value:
    name: "KO_DATA_PATH"
    value: "/tmp/knative/"
- command: update
  path: spec.install.spec.deployments.(name==knative-operator).spec.template.spec.containers.(name==knative-operator).volumeMounts[+]
  value:
    name: "serving-manifest"
    mountPath: "/tmp/knative/knative-serving/0.17.2"
- command: update
  path: spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes[+]
  value:
    name: "serving-manifest"
    configMap:
      name: "ko-data-serving"
      items:
        key: "knative-serving-ci.yaml"
        path: "knative-serving-ci.yaml"
```

The main benefit I see is that that's a lot more maintainable imo and easier to see what's being done. To apply directly, you can do:

```
cat << EOF | yq write --inplace --script - "$1"
- command: update
  path: spec.install.spec.deployments.(name==knative-operator).spec.template.spec.containers.(name==knative-operator).env[+]
  value:
    name: "KO_DATA_PATH"
    value: "/tmp/knative/"
- command: update
  path: spec.install.spec.deployments.(name==knative-operator).spec.template.spec.containers.(name==knative-operator).volumeMounts[+]
  value:
    name: "serving-manifest"
    mountPath: "/tmp/knative/knative-serving/0.17.2"
- command: update
  path: spec.install.spec.deployments.(name==knative-operator).spec.template.spec.volumes[+]
  value:
    name: "serving-manifest"
    configMap:
      name: "ko-data-serving"
      items:
        key: "knative-serving-ci.yaml"
        path: "knative-serving-ci.yaml"
EOF
```

Reply:
Sounds good. I will try to change to use the script soon.